src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -174,6 +174,9 @@ export class TabsTitleControl extends TitleControl {
 		this.tabsContainer.setAttribute('role', 'tablist');
 		this.tabsContainer.draggable = true;
 		this.tabsContainer.classList.add('tabs-container');
+		this.tabsContainer.addEventListener('mouseenter', this.fixTabWidths.bind(this));
```

Comment:
Could we have the listeners only be there when we need them based on the setting?